pkg/client/apiutil/dynamicrestmapper.go

```diff
@@ -94,25 +95,38 @@ func WithCustomMapper(newMapper func() (meta.RESTMapper, error)) DynamicRESTMapp
 	}
 }
 
+// WithDiscoveryFilter sets a groupVersion filter for the discovery client
+// that allows to limit the relevant groups for the mapper
+func WithDiscoveryFilter(filter func(groupVersion string) bool) DynamicRESTMapperOption {
+	return func(drm *dynamicRESTMapper) error {
+		drm.discoveryClient = &FilterDiscoveryClient{
+			Discovery: drm.discoveryClient,
+			Filter:    filter,
+		}
+		return nil
+	}
+}
+
 // NewDynamicRESTMapper returns a dynamic RESTMapper for cfg. The dynamic
 // RESTMapper dynamically discovers resource types at runtime. opts
 // configure the RESTMapper.
 func NewDynamicRESTMapper(cfg *rest.Config, opts ...DynamicRESTMapperOption) (meta.RESTMapper, error) {
-	client, err := discovery.NewDiscoveryClientForConfig(cfg)
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
 	if err != nil {
 		return nil, err
 	}
 	drm := &dynamicRESTMapper{
+		discoveryClient: discoveryClient,
 		limiter: &dynamicLimiter{
 			rate.NewLimiter(rate.Limit(defaultRefillRate), defaultLimitSize),
 		},
-		newMapper: func() (meta.RESTMapper, error) {
-			groupResources, err := restmapper.GetAPIGroupResources(client)
-			if err != nil {
-				return nil, err
-			}
-			return restmapper.NewDiscoveryRESTMapper(groupResources), nil
-		},
+	}
+	drm.newMapper = func() (meta.RESTMapper, error) {
+		groupResources, err := restmapper.GetAPIGroupResources(drm.discoveryClient)
```

Comment:
add a comment emphasizing that we're using the struct's `drm.discoveryClient` and not just `discoveryClient` because it might get overridden below

Reply:
done